File: util/compare/fact.js
function хэвлэ(тоо) {
  console.log(тоо);
}

function факториал(н) {
  if (н <= 1) {
    return 1;
  }

  const бодсон = н * факториал(н - 1);
  хэвлэ(бодсон);
  return бодсон;
}

function үндсэн() {
  return факториал(12); // 12! = 479001600
}

// Measure execution time
const start = performance.now();
const result = үндсэн();
const end = performance.now();

console.log("Final result:", result);
console.log(`Execution time: ${(end - start).toFixed(5)} milliseconds`);

File: util/compare/fact.c
#include <stdio.h>
#include <time.h>

void print_result(long result) {
    printf("%ld\n", result);
}

long factorial(int n) {
    if (n <= 1) {
        return 1;
    }

    long result = n * factorial(n - 1);
    print_result(result);
    return result;
}

int main() {
    // Measure execution time
    clock_t start = clock();

    long result = factorial(12);  // 12! = 479001600

    clock_t end = clock();
    double elapsed_time = (double)(end - start) / CLOCKS_PER_SEC;

    printf("Final result: %ld\n", result);
    printf("Execution time: %.9f seconds\n", elapsed_time);

    return 0;
}
